hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/TestMORDataSource.scala

```diff
@@ -541,6 +541,66 @@ class TestMORDataSource extends HoodieSparkClientTestBase with SparkDatasetMixin
     assertEquals(0, hudiSnapshotDF3.count()) // 100 records were deleted, 0 record to load
   }
 
+  @Test
+  def testDeletesWithLowerOrderingValue() : Unit = {
```

Comment:
Should this test be added to `TestRecordLevelIndex` since it's record index specific?